internal/campaigns/run_steps.go

```diff
@@ -271,3 +335,169 @@ func (e stepFailedErr) SingleLineError() string {
+	PreviousStep StepResult
+	Repository   graphql.Repository
+}
+
+// ToFuncMap returns a template.FuncMap to access fields on the StepContext in a
+// text/template.
+func (stepCtx *StepContext) ToFuncMap() template.FuncMap {
+	return template.FuncMap{
+		"join": func(list []string, sep string) string {
+			return strings.Join(list, sep)
+		},
+		"previous_step": func() map[string]interface{} {
+			result := map[string]interface{}{
+				"modified_files": stepCtx.PreviousStep.ModifiedFiles(),
+				"added_files":    stepCtx.PreviousStep.AddedFiles(),
+				"deleted_files":  stepCtx.PreviousStep.DeletedFiles(),
+			}
+
+			if stepCtx.PreviousStep.Stdout != nil {
+				result["stdout"] = stepCtx.PreviousStep.Stdout.String()
+			} else {
+				result["stdout"] = ""
+			}
+
+			if stepCtx.PreviousStep.Stderr != nil {
+				result["stderr"] = stepCtx.PreviousStep.Stderr.String()
+			} else {
+				result["stderr"] = ""
+			}
+
+			return result
+		},
+		"repository": func() map[string]interface{} {
+			return map[string]interface{}{
+				"search_result_paths": stepCtx.Repository.SearchResultPaths(),
+				"name":                stepCtx.Repository.Name,
+			}
+		},
+	}
+}
+
+// StepResult represents the result of a previously executed step.
+type StepResult struct {
+	// Files are the changes made to files by the step.
+	Files StepChanges
+
+	// Stdout is the output produced by the step on standard out.
+	Stdout *bytes.Buffer
+	// Stderr is the output produced by the step on standard error.
+	Stderr *bytes.Buffer
+}
+
+// StepChanges are the changes made to files by a previous step in a repository.
+type StepChanges struct {
+	Modified []string
+	Added    []string
+	Deleted  []string
+}
+
+// ModifiedFiles returns the files modified by a step, whitespace-separated in a single string.
+func (r StepResult) ModifiedFiles() string {
+	if len(r.Files.Modified) == 0 {
+		return ""
+	}
+	return strings.Join(r.Files.Modified, " ")
+}
+
+// AddedFiles returns the files added by a step, whitespace-separated in a single string.
+func (r StepResult) AddedFiles() string {
+	if len(r.Files.Added) == 0 {
+		return ""
+	}
+	return strings.Join(r.Files.Added, " ")
+}
+
+// DeletedFiles returns the files deleted by a step, whitespace-separated in a single string.
+func (r StepResult) DeletedFiles() string {
+	if len(r.Files.Deleted) == 0 {
+		return ""
+	}
+	return strings.Join(r.Files.Deleted, " ")
+}
+
+func parseGitStatus(out []byte) (StepChanges, error) {
+	result := StepChanges{}
+
+	stripped := strings.TrimSpace(string(out))
+	if len(stripped) == 0 {
+		return result, nil
+	}
+
+	for _, line := range strings.Split(stripped, "\n") {
+		if len(line) < 4 {
+			return result, fmt.Errorf("git status line has unrecognized format: %q", line)
+		}
+
+		file := line[3:]
+
+		switch line[0] {
+		case 'M':
+			result.Modified = append(result.Modified, file)
+		case 'A':
+			result.Added = append(result.Added, file)
+		case 'D':
+			result.Deleted = append(result.Deleted, file)
+		}
```

Comment:
It's a niche case, and probably irrelevant for an MVP, but would we also want to provide ways to access files that are renamed or copied as well by checking for [`R` and `C`, respectively](https://git-scm.com/docs/git-status#_short_format)?

Reply:
No, you're right. I'll do that. I simply forgot about those.

Reply:
I added renames but I didn't add `C`, because ... I couldn't even get git to show a copied file: https://stackoverflow.com/questions/27198414/git-status-git-diff-c-not-detecting-file-copy-in-index And I would assume that other users then also expect new files to show up under `added` and not under copied.

If we get feedback to the contrary, we can add it.